webxr/openxr/mod.rs

```diff
@@ -1103,16 +1106,44 @@ impl DeviceAPI for OpenXrDevice {
             }
         }
 
-        let mut guard = self.shared_data.lock().unwrap();
-        let mut data = guard.as_mut().unwrap();
-        let frame_state = match self.frame_waiter.wait() {
-            Ok(frame_state) => frame_state,
-            Err(e) => {
-                error!("Error waiting on frame: {:?}", e);
-                return None;
+        let (frame_state, secondary_state) = if self
```

Comment:
i wonder if a `secondary_enabled` can be stored on `self` instead: it's readonly

Reply:
Problem is it's needed by the `views()` function, so we'd end up duplicating it.

Reply:
Yes, but it's a boolean, so it's cheap and better than locking twice

Reply:
That's true, this double-locking is annoying.

Reply:
Fixed.